### DISCUSSION TITLE
Is it planned to be able to share location access between users?

### DISCUSSION THREAD

**Discussion by samtate** [Freika/dawarich#134] [Ideas]
Will there be a feature where you can share location to a friend or family who also has an account on the dawarich server?

**Comment by Freika** [comment]
At some point there might be this kind of functionality, yes. Not in the nearest future, though

**Reply by epoch-philosophy** [reply]
Genuinely, for most self-hosted users/people, what is the point of something like Dawarich if not the ability to share locations privately with friends or family?

**Reply by Freika** [reply]
There wouldn't be as many Dawarich users if it would be the only point

**Reply by epoch-philosophy** [reply]
I understand that certainly couldn't be the only feature, but with multi-user support I was just about certain the ability to see other user's location data on Darawhich was present. Didn't realize it had this limitation. Hopefully that's a feature that can come sooner than later! Would be an important one for people looking to get off iCloud's FindMy and Life365.

**Reply by Freika** [reply]
I have it planned, so Dawarich will have it :)

**Reply by epoch-philosophy** [reply]
Awesome. Would be the very feature that brings me (and family) over! Been looking at a privacy friendly location sharing service!

**Reply by n7ekb** [reply]
> I understand that certainly couldn't be the only feature, but with multi-user support I was just about certain the ability to see other user's location data on Darawhich was present. Didn't realize it had this limitation. Hopefully that's a feature that can come sooner than later! Would be an important one for people looking to get off iCloud's FindMy and Life365.

It seems that the "Family" feature addresses this issue.  Sort of.  The Family layer on the map however is default "off" with no apparent user-configurable options to persist an "on" setting.  It's really hard to convince non-technical Life365 family members to use a "better tool" that requires tweaking every time they refresh their browser window.

**Reply by k8ieone** [reply]
The map layer settings *should* be persisted. It's a bug. See https://github.com/Freika/dawarich/discussions/1872#discussioncomment-14845525

**Comment by eo2day** [comment]
+1 to this. I brought this capability up yesterday. My current use case is to have a self-hosted Life-360 alternative. The built-in user management for Dawarich is great! A group location sharing feature would be awesome!

The ability to create and join groups to share current locations (and possibly historic data for the length of time they've been in the group) would be a great security feature. It would allow a hosted server to reach a wider group of people than the default OwnTracks Recorder/Frontend, since the data would not be visible to every member that can access the UI. A host could add their whole extended family to the server, and each immediate family group could share with each other. A broader-reaching group could be set up for events, traveling, and other situations where you might want to be able to locate others.

**Comment by samtate** [comment]
It looks like owntracks natively supports user location sharing if the server uses an MQTT endpoint. So perhaps this could be achieved by adding the MQTT endpoint capabity to dawarich in future releases and letting the Owntracks app be the Frontend for user sharing

**Reply by eo2day** [reply]
Heyo!
For my current usage and the scenario I proposed, I don't think that the Owntracks Frontend is quite what would work. That's currently what I'm using, and while it works for close people you are okay with having 24/7 access to everyone else's location, it's not great for supporting a greater range of people who may want to share locations among their smaller groups. The specific scenario I'm thinking of is sharing one node with my wife, parents, and grandparents on each side. If I only wanted to share my location with my wife, and not anyone else in that set, then the Owntracks Frontend won't work since it automatically displays every person. Or, I might have a friend couple that want to share locations with each other, but not my grandparents.

Now, if the owntracks frontend could be adapted to display a group's set of locations, that would be pretty neat, but I think it would need each group to have it's own instance of the owntracks recorder? Perhaps I'm missing something on that front, since I'm still fairly new to this tech.

**Reply by ajain-93** [reply]
It really depends on what system you are using for reporting the location to the dawarich backend. The integration that I am using is Home Assistant. The HA companion app reports the device location to my selfhosted HA instance, which then forwards the new location to Dawarich. In our case we are using HA to show the current location of all family members since all devices have the HA app.

**Comment by Baloe2000** [comment]
This would be a very welcome feature. It would make dawarich not only a timeline replacement but also a replacement for life360 and other privacy invading location sharing services like these. Of which there are no really solid self hosted replacements for yet. Hoping this could be intergrated sometime! Thanks to the developers for putting your efforts in Dawarich.

**Comment by kronn** [comment]
Just to stick to the naming of the app, I would call the location-sharing of the current location "dabinich". I my mind, it would only share the last/current location of that user. If dawarich also adds a group functionality, this would become "dasindwir". But even while typing this, I can feel the goal-post moving deamon of feature-creep grinning behind my NAS.

That being said, I think this could make a great contribution-opportunity, but I would want to have this designed (as in described, architected) by @Freika, so that it fits the general feel and idea of the app. The simplest solution I see would be:

- no explicit groups as the self-hosted instance itself is a grouping
- a map that shows the current/last known location of all users
- maybe: a setting in the user to NOT be part of this

But I do not know the codebase well enough to judge if this really is the best starting point.

For other use-cases, the HA-supported-way might be better. For simple stand-alone use-cases, I propose this here.

**Reply by pwwukpw** [reply]
Any update on being able to see  on map other users on same instance / server ? 

Many thanks Peter

**Reply by Freika** [reply]
Still hope to release it by the end of this year, may be earlier

**Comment by Freika** [comment]
@samtate  @epoch-philosophy @eo2day @ajain-93 @Baloe2000 @kronn @pwwukpw @greccode @haztecaso @adriy-be and others:

FYI 0.34.0 (RC branch, docker image `freikin/dawarich:rc`) release contains the Family feature, available for testing (image only built for AMD64 arch)

**Comment by joschreiner** [comment]
@Freika with the new public link sharing system, would it be possible to implement a feature with which one can add people just based on the share link and secret? This would also allow for location sharing between instances?

**Reply by Freika** [reply]
There is no plans for sharing location between instances.
But it's possible later on link-based location shared for groups will be introduced
